models/ed/R/write_restart.ED2.R

```diff
@@ -128,26 +128,19 @@ write_restart.ED2 <- function(outdir,
 
   ##### Modify ED2IN
   ed2in_path <- file.path(rundir, runid, "ED2IN")
-  ed2in_orig <- readLines(ed2in_path)
-  ed2in_new <- ed2in_orig
-
-  ## IED_INIT_MODE = 5 --> Run from history.h5 file
-  tag_val_list <- list("RUNTYPE" = "history",
-                       "IED_INIT_MODE" = 5,
-                       "SFILIN" = file.path(mod_outdir, runid, "history"),
-                       "IMONTHA" = strftime(start.time, "%m"), 
-                       "IDATEA" = strftime(start.time, "%d"),
-                       "IYEARA" = strftime(start.time, "%Y"),
-                       "ITIMEA" = strftime(start.time, "%H%M"),
-                       "IMONTHZ" = strftime(stop.time, "%m"),
-                       "IDATEZ" = strftime(stop.time, "%d"),
-                       "IYEARZ" = strftime(stop.time, "%Y"),
-                       "ITIMEZ" = strftime(stop.time, "%H%M"))
-
-  modstr <- 'Modified by write.restart.ED2'
-  ed2in_new <- ed2in_set_value_list(tag_val_list, ed2in_orig, modstr)
-
-  writeLines(ed2in_new, file.path(ed2in_path))
+  ed2in_orig <- read_ed2in(ed2in_path, check = FALSE)
+
+  ed2in_new <- modify_ed2in(
+    ed2in_orig,
+    start_date = start.time,
+    end_date = stop.time,
+    RUNTYPE = "HISTORY",
+    IED_INIT_MODE = 5,
```

Comment:
I'm asking this out of curiosity, why is this 5 but not 4?

Reply:
Oof, I did this ages ago, back at the Berkeley hackathon, and I don't remember why I picked this exact value. Maybe because it's more flexible than 4? The ED2IN documentation says this:

```
   !    4.  Same as 1, but using ED2.1 H5 history/state files that take the form:          !
   !        'dir/prefix-gxx.h5'                                                            !
   !        Initialization files MUST end with -gxx.h5 where xx is a two digit integer     !
   !        grid number.  Each grid has its own initialization file.  As an example, if a  !
   !        user has two files to initialize their grids with:                             !
   !        example_file_init-g01.h5 and example_file_init-g02.h5                          !
   !        NL%SFILIN = 'example_file_init'                                                !
   !                                                                                       !
   !    5.  This is similar to option 4, except that you may provide several files         !
   !        (including a mix of regional and POI runs, each file ending at a different     !
   !        date).  This will not check date nor grid structure, it will simply read all   !
   !        polygons and match the nearest neighbour to each polygon of your run.  SFILIN  !
   !        must have the directory common to all history files that are sought to be used,! 
   !        up to the last character the files have in common.  For example if your files  !
   !        are                                                                            !
   !        /mypath/P0001-S-2000-01-01-000000-g01.h5,                                      !
   !        /mypath/P0002-S-1966-01-01-000000-g02.h5,                                      !
   !        ...                                                                            !
   !        /mypath/P1000-S-1687-01-01-000000-g01.h5:                                      !
   !        NL%SFILIN = '/mypath/P'                                                        !
```

Reply:
yeah as we only have one grid (g001, is that correct?) and `N_POI =1` I was wondering if ours is case-4 and selecting 5 would confuse restart, but maybe it doesn't matter..anyway, I'll be testing it soon

Reply:
Makes sense. The last time I touched this code was when I first developed it, and it _seemed_ to do what I wanted then, but I don't think I ever tried to do any real state data assimilation. Feel free to change it to whatever makes it work (or might even make sense to tweak `write_restart` to take it as an argument).

Reply:
I though we always use INIT mode 3? @mdietze

Reply:
My understanding is that we need 4 or 5 for state data assimilation, so ED reads more information about the last state from the history files.